lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -320,4 +282,51 @@ sub destroy {
     $self->render(json => {result => int($ret)});
 }
 
+=over 4
+
+=item prepare_settings()
+
+Internal method to prepare settings when add or update admin table.
+Use by both B<create()> and B<update()> method.
+
+=back
+
+=cut
+
+sub prepare_settings {
+    my ($self, $table, $entry) = @_;
+    my $validation = $self->validation;
+
+    for my $par (@{$tables{$table}->{required}}) {
+        $validation->required($par);
+        if (!defined $validation->param($par)) {
+            next;
+        }
+        $entry->{$par} = trim $validation->param($par);
+    }
+
+    if ($validation->has_error) {
+        my $error = "wrong parameter: ";
+        for my $par (@{$tables{$table}->{required}}) {
+            $error .= " $par" if $validation->has_error($par);
+        }
+        $self->render(json => {error => $error}, status => 400);
+        return;
```

Comment:
This seems problematic. An error is rendered, not sure what will be returned here, and `create` will probably continue with no settings?

I'd find it cleaner to `die` here, `catch` it along with the database call and only render in the consuming function.

Reply:
If the `return;` is a `return undef;` and the consuming function handles the `undef`-case correctly I'm actually fine with this.

Reply:
Then you can't pass the error message along though. But have to agree that this needs an explicit return value the caller can handle appropriately.

Reply:
I remove return self-render in this inner function, only return an error message when check parameter failed.